pkgs/ok_http/lib/src/ok_http_client.dart

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// An Android Flutter plugin that exposes the [OkHttp](https://square.github.io/okhttp/) HTTP client.
+///
+/// The platform interface must be initialized before using this plugin e.g. by
+/// calling
+/// [`WidgetsFlutterBinding.ensureInitialized`](https://api.flutter.dev/flutter/widgets/WidgetsFlutterBinding/ensureInitialized.html)
+/// or
+/// [`runApp`](https://api.flutter.dev/flutter/widgets/runApp.html).
+library;
+
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:http/http.dart';
+import 'package:jni/jni.dart';
+
+import 'third_party/okhttp3/_package.dart' as bindings;
+
+/// An HTTP [Client] utilizing the [OkHttp](https://square.github.io/okhttp/) client.
+///
+/// Example Usage:
+/// ```
+/// void main() async {
+///   var client = OkHttpClient();
+///   final response = await client.get(
+///       Uri.https('www.googleapis.com', '/books/v1/volumes', {'q': '{http}'}));
+///   if (response.statusCode != 200) {
+///     throw HttpException('bad response: ${response.statusCode}');
+///   }
+///
+///   final decodedResponse =
+///       jsonDecode(utf8.decode(response.bodyBytes)) as Map;
+///
+///   final itemCount = decodedResponse['totalItems'];
+///   print('Number of books about http: $itemCount.');
+///   for (var i = 0; i < min(itemCount, 10); ++i) {
+///     print(decodedResponse['items'][i]['volumeInfo']['title']);
+///   }
+/// }
+/// ```
+///
+class OkHttpClient extends BaseClient {
+  late bindings.OkHttpClient _client;
+
+  OkHttpClient() {
+    _client = bindings.OkHttpClient.new1();
+  }
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    var requestUrl = request.url.toString();
+    var requestHeaders = request.headers;
+    var requestMethod = request.method;
+    var requestBody = await request.finalize().toBytes();
+
+    // A Completer to return the response, as OkHttp's enqueue API is async
+    final responseCompleter = Completer<StreamedResponse>();
+
+    // Create an OkHttp Request Builder
+    var reqBuilder = bindings.Request_Builder().url1(requestUrl.toJString());
+
+    requestHeaders.forEach((headerName, headerValue) {
+      reqBuilder.addHeader(headerName.toJString(), headerValue.toJString());
+    });
+
+    // OkHttp doesn't allow a non-null RequestBody for GET and HEAD requests
+    // So, we need to handle this case separately
+    bindings.RequestBody okReqBody;
+    if (requestMethod != 'GET' && requestMethod != 'HEAD') {
+      okReqBody = bindings.RequestBody.create10(requestBody.toJArray());
+    } else {
+      okReqBody = bindings.RequestBody.fromReference(jNullReference);
```

Comment:
I think we do, since a couple of lines later we have:
```dart
reqBuilder.method(
  requestMethod.toJString(),
  okReqBody,
);
```
Of definition
```dart
Request_Builder method(JString string, RequestBody requestBody)
```
Here, `okReqBody` cannot be null or left uninitialized. Moreover, it has to be of the given type. The only workaround I could find for this was to make create a `RequestBody` object from a null pointer (Essentially a null object when it operates with Kotlin code)

Reply:
OK